src/filesystem/FileSystem.c

```diff
@@ -19,6 +18,7 @@ void filesystemInit(flashConfiguration_t *flashConfig,
     filesystemConfig->filesystemStartSector = 1019;
     filesystemConfig->filesystemEndSector = 1023;
     filesystemConfig->flash = flashConfig;
+    filesystemConfig->numberOfFreeSectors = 1019;
 
```

Comment:
Where do these numbers come from btw? Can you find a way to make that clearer? E.g., using object like macros or enums?

Reply:
Overall this is quite a long function. Do you see reasonable ways to split it into smaller steps? Maybe at the points where comments are used for explanation atm?